bundles/CoreBundle/DependencyInjection/Compiler/EntityModelFactoryPass.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\DependencyInjection\Compiler;
+
+use Pimcore\Model\Asset;
+use Pimcore\Model\DataObject;
+use Pimcore\Model\Document;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\HttpKernel\Profiler\Profiler;
+
+/**
+ * @internal
+ */
+final class EntityModelFactoryPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $classes = [
+            Asset::class,
+            Asset\Archive::class,
+            Asset\Audio::class,
+            Asset\Folder::class,
+            Asset\Image::class,
+            Asset\Text::class,
+            Asset\Unknown::class,
+            Asset\Video::class,
+
+            Document::class,
+            Document\Email::class,
+            Document\Folder::class,
+            Document\Hardlink::class,
+            Document\Link::class,
+            Document\Newsletter::class,
+            Document\Page::class,
+            Document\Printcontainer::class,
+            Document\Printpage::class,
+            Document\Snippet::class,
+
+            DataObject\Folder::class,
+        ];
```

Comment:
I think this should be a `const`.

Reply:
I'm planning to go for a generic approach anyway ... so basically scanning the classes in `model/` and initializing them.